src/main/java/org/apache/commons/io/input/QueueInputStream.java

```diff
@@ -57,24 +59,45 @@
  */
 public class QueueInputStream extends InputStream {
 
-    private final BlockingQueue<Integer> blockingQueue;
+    final BlockingQueue<Integer> blockingQueue;
 
     /**
-     * Constructs a new instance with no limit to its internal buffer size.
+     * Constructs a new instance with no limit to its internal queue size
      */
     public QueueInputStream() {
         this(new LinkedBlockingQueue<>());
     }
 
     /**
-     * Constructs a new instance with given buffer
+     * Constructs a new instance with given queue
      *
      * @param blockingQueue backing queue for the stream
      */
     public QueueInputStream(final BlockingQueue<Integer> blockingQueue) {
         this.blockingQueue = Objects.requireNonNull(blockingQueue, "blockingQueue");
     }
 
+    /**
```

Comment:
I'd rather not have these factory methods, if we need something to build, let's use a Builder.

Reply:
@maxxedev Hm, let me try something and get back to you...